### ISSUE TITLE
Feature Request: Text wrapping/scaling on smaller screens

### ISSUE THREAD

**Issue by tgm-git** [lemmy-gtk/lemoa#27]
# Feature Request: Smaller screens
I am using lemoa on my Librem 5 phone. Everything seems to work with the exception of the post titles and the more pages buttons being wider than my screen. 

Ideally the posts and more button would scale up to the screen width. My phone screen is 720x1440 and I view lemoa in portrait mode.
![img](https://github.com/lemmygtk/lemoa/assets/4103745/cadd8d73-fea6-4ec5-89d0-4f2e7237fd09)

**Comment by Shatur** [comment]
Browser version for the reference (works great from browser, btw):
![image_2023-09-14_19-25-26](https://github.com/lemmygtk/lemoa/assets/22453358/d6ec9291-a61e-4dac-affd-6e0d8e5df1f6)

**Comment by Bnyro** [comment]
Well, Gtk is not comparable to a web app ...

**Comment by Shatur** [comment]
Fair point. But I have another app for the reference: [Headlines](https://gitlab.com/caveman250/Headlines).
It's a Reddit client, but it worked great on my screen.

**Comment by Bnyro** [comment]
We're now decreasing the image sizes based on the screen size, and use libadwaita by default which should have improved things.

Could you please build the app from source from the latest commit and give an update on the situation? (I don't have any small screen Linux device I could use for testing unfortunately)

**Comment by Bnyro** [comment]
It's cool to see some users of Linux phones, I didn't know that's actually a thing :)

**Comment by Shatur** [comment]
> Could you please build the app from source from the latest commit

Thanks, compiling!

**Comment by Shatur** [comment]
Better, but still huge because of the buttons. Also for some reason images are loading slow.
![image_2023-09-14_21-02-11](https://github.com/lemmygtk/lemoa/assets/22453358/a56f8460-afe5-4a1f-a905-1a695703c751)

**Comment by Shatur** [comment]
In QML buttons that didn't fit automatically moved into sandwich menu.
But the most issue imho with posts buttons. They are also too big. May I suggest to try to use approach [similar to Headlines](https://gitlab.com/caveman250/Headlines/raw/master/screenshots/headlines-3.png)?

**Comment by Bnyro** [comment]
The approach is to add at most two items to a line and order them vertically as well as replacing buttons with simple text fields?

**Comment by Shatur** [comment]
Yes. I think that it would even look a bit better, I like Headlines interface a lot.

**Comment by TheKrafter** [comment]
> The buttons in the header bar should be easy to fix - we can just use a dropdown menu on small screens.
> 
> However, I really have no idea why libadwaita doesn't automatically use small buttons for small screens. We should be able to wrap the overflowing buttons into a new line however.

It's because GTK4/Libadwaita isn't designed to have a bunch of these kinds of buttons across the screen. I would take a look at the [GNOME HIG Patterns](https://developer.gnome.org/hig/patterns.html), and (as many others have mentioned) have a look at how headlines' UI works.

If the mobile-centric UI doesn't look great on desktop, I'd look into Adw.Breakpoint (I don't have a great resource for you, but maybe check out the [Migrating to Breakpoints](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1-latest/migrating-to-breakpoints.html) page on the docs).
